### PR TITLE
Event Stream IO layer

### PR DESCRIPTION
**Pull request by exodist**

### CODE REVIEW

File: lib/Test2/API.pm [hunk 1]
```diff
@@ -694,6 +711,26 @@ These exports lack the 'test2_' prefix because of how important/common they
...
+
+    my $stdout = test2_stdout();
+    my $stderr = test2_stderr();
+
+    print $stdout "This goes to STDOUT\n";
+    print $stderr "This goes to STDOUT\n";
+
+It is always possible that code you are testing could use C<select> or replace
+STDOUT and STDERR. Test2 duplicates STDOUT and STDERR during initialization to
+insure you always have access to the real ones. L<Test2::Formatter> classes
```
Comment (autarch) [inline lib/Test2/API.pm]: ensure

File: lib/PerlIO/via/Test2.pm [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+=item stream_name => $string
+
+Name of the stream (whatever you want, but must be unique).
+
+=item diagnostics => $bool
+
+If this is true then C<diagnostics> will be set to true on the
+L<Test2::Event::Output> events generated. This effectively is a switch for the
+formatter to know if the even should be rendered to STDOUT or STDERR for
```
Comment (autarch) [inline lib/PerlIO/via/Test2.pm]: "if the event"

**Comment by haarg** [comment]
Does this really belong in the Test-Simple dist?

**Comment by exodist** [comment]
@haarg I have been debating that myself. It started in Test2::Harness, then I moved it up to Test2::Suite, most recently I moved it to Test2. I might end up moving it back down, I am trying to find the place that makes the most sense.

I feel like the PerlIO part does belong in Test2, including the Test2::API command to put it on a handle. However I think the plugins and bigger tools that use it do not belong in Test-Simple.

**Comment by exodist** [comment]
I am gonna put this in another dist, not sure which yet, but not test-simple.

**Pull request closed without merging**